pkg/applier/applier.go

```diff
@@ -500,6 +644,87 @@ func (a *Applier) Update(
 
 }
 
+//Delete deletes an unstructured object.
+func (a *Applier) Delete(
+	u *unstructured.Unstructured,
+) error {
+
+	klog.V(2).Info("Delete: ",
+		" Kind: ", u.GetKind(),
+		" Name: ", u.GetName(),
+		" Namespace: ", u.GetNamespace())
+	var clientDeleteOptions []client.DeleteOption
+	if a.applierOptions != nil {
+		clientDeleteOptions = a.applierOptions.ClientDeleteOption
+	}
+	deleteOptions := &client.DeleteOptions{}
+	clientDeleteOption := deleteOptions.ApplyOptions(clientDeleteOptions)
+	c := a.client
+	if a.applierOptions.DryRun {
+		printUnstructure(u)
+		c = client.NewDryRunClient(c)
+	}
+	err := retry.OnError(*a.applierOptions.Backoff, func(err error) bool {
+		if err != nil && !errors.IsNotFound(err) {
+			klog.V(2).Infof("Retry delete %s", err)
+			return true
+		}
+		return false
+	}, func() error {
+		err := c.Delete(context.TODO(), u, clientDeleteOption)
+		if err != nil {
+			klog.V(2).Infof("Error while deleting %s", err)
+		}
+		return err
+	})
+	if err != nil && !errors.IsNotFound(err) {
+		klog.V(2).Info("Unable to delete:", "Error", err,
+			" Kind: ", u.GetKind(),
+			" Name: ", u.GetName(),
+			" Namespace: ", u.GetNamespace())
+		return err
+	}
+	if a.applierOptions.ForceDelete &&
+		u.GetKind() != reflect.TypeOf(apiextensions.CustomResourceDefinition{}).Name() {
```

Comment:
so not force deleting crd, that make sense, and how about namespace?

Reply:
and it would be nice to doc in either the flags' help message?

Reply:
added in the doc